plugins/inputs/nginx_upstream_check/nginx_upstream_check.go

```diff
@@ -0,0 +1,184 @@
+package nginx_upstream_check
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type NginxUpstreamCheck struct {
+	Urls            []string
+	ResponseTimeout internal.Duration
+
+	tls.ClientConfig
+
+	client *http.Client
+}
+
+type NginxUpstreamCheckData struct {
+	Servers struct {
+		Total      uint64                     `json:"total"`
+		Generation uint64                     `json:"generation"`
+		Server     []NginxUpstreamCheckServer `json:"server"`
+	} `json:"servers"`
+}
+
+type NginxUpstreamCheckServer struct {
+	Index    uint64 `json:"index"`
+	Upstream string `json:"upstream"`
+	Name     string `json:"name"`
+	Status   string `json:"status"`
+	Rise     uint64 `json:"rise"`
+	Fall     uint64 `json:"fall"`
+	Type     string `json:"type"`
+	Port     uint16 `json:"port"`
+}
+
+var sampleConfig = `
+  ## An array of nginx_upstream_check URLs
+  ## They should be set to return a JSON formatted response
+  urls = ["http://localhost/status?format=json"]
+
+  ## HTTP response timeout (default: 5s)
+  # response_timeout = "5s"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+var description = "Read nginx_upstream_check module status information (https://github.com/yaoweibin/nginx_upstream_check_module)"
+
+func (check *NginxUpstreamCheck) SampleConfig() string {
+	return sampleConfig
+}
+
+func (check *NginxUpstreamCheck) Description() string {
+	return description
+}
+
+func (check *NginxUpstreamCheck) Gather(accumulator telegraf.Accumulator) error {
+	var waitGroup sync.WaitGroup
+
+	if check.client == nil {
+		client, error := check.createHTTPClient()
+		if error != nil {
```

Comment:
It's better to rename `error` variable name into `err` one. since `error` is a type name. And `err` is de-facto name in Go codebase.